src/services/jsTyping.ts

```diff
@@ -192,7 +192,7 @@ namespace ts.JsTyping {
             const cleanedTypingNames = map(inferredTypingNames, f => f.replace(/((?:\.|-)min(?=\.|$))|((?:-|\.)\d+)/g, ""));
 
             if (safeList !== EmptySafeList) {
-                mergeTypings(filter(cleanedTypingNames, f => safeList.has(f)));
+                mergeTypings(map(filter(cleanedTypingNames, f => safeList.has(f)), (f => safeList.get(f))));

```

Comment:
This looks like a good candidate to use `ts.mapDefined(cleanedTypingNames, f => safeList.get(f))`.